### DISCUSSION TITLE
Single grain extraction

### DISCUSSION THREAD

**Discussion by sajutabraham** [neperfepx/neper#350] [Q&A] [answered]
I would like to visualize a **single grain** (in 3D) from a tessellated microstructure, before and after meshing. Can somebody help, please?

**Comment by rquey** [comment, marked as answer]
Use `-showcell id==...` for the tessellation ("before meshing") and `-showelset id=..." for the mesh ("after meshing").  Here is an example:

```
neper -T -n 10 -reg 1 -o n10
neper -M n10.tess
neper -V n10.tess,n10.msh -dataedgerad 0.003 -showtess 1 -showmesh 0 -showcell "id==3" -print before -showtess 0 -showmesh 1  -dataelt1drad 0.003 -dataelt3dedgerad 0.0015 -showelset "id==3"  -showelt1d elt3d_shown -print after
```
or, more simply,
```
neper -V n10.tess,n10.msh -dataelt1drad 0.003 -showelset "id==3" -showelt1d elt3d_shown -dataelt3dedgerad 0 -print before -dataelt3dedgerad 0.0015 -print after
```

**Reply by sajutabraham** [reply]
Thank you very much. But it seems to be a difficult task if the number of grains in a polycrystal is more and we are interested in visualizing only grains with some particular shapes.

**Reply by rquey** [reply]
See the documentation of option `-showelset`, which can take various arguments. This is a highly customizable option. For example, to view all cells that have a sphericity higher than 0.85, use `-showelset "sphericity>0.85"`, etc. You can also specify which elsets to show using `-showelset "file(...)"`.

**Reply by sajutabraham** [reply]
Thank you very much!
